README.md
Breathe est un thème pour Dotclear2.

Depuis "Personnaliser le thème", il est possible de choisir :
* le type de menu à afficher (Simplemenu ou [Menu] (http://plugins.dotaddict.org/dc2/details/menu))
* un environnement coloré prédéfini (si le thème rencontre un minimum de succès, je m'y attèlerai...)
* d'afficher ou non un dock au-dessus du footer qui liste les 25 derniers billets ayant une image (le plugin [listImages] (http://plugins.dotaddict.org/dc2/details/listImages) est requis)
* d'afficher en page d'accueil deux types de slides avec ou sans vignettes (le plugin [listImages] (http://plugins.dotaddict.org/dc2/details/listImages) est requis)
* d'afficher ou non le slide sélectionné dans les pages suivantes de la page d'accueil.

Dans le contexte billet seul, le service [addthis](http://www.addthis.com/) a été inclus pour faciliter le partage (fichier user_share.html) et la navigation se fait dans la catégorie et non sur la totalité du blog.

Ce thème n'est pas compatible avec Internet Explorer 8 et antérieurs.

Le thème
--------

Titre du blog, un simple lien ou une image cliquable.

Le fichier _top.html est prévu pour afficher un simple lien vers votre blog :
```html
<h1><span><a href="{{tpl:BlogURL}}">{{tpl:BlogName encode_html="1"}}</a></span></h1>
```
ou une image liée (/breathe/img/logo.png) :
```html
<h1><a href="{{tpl:BlogURL}}"><img src="{{tpl:BlogThemeURL}}/img/logo.png" alt="{{tpl:BlogName encode_html="1"}}" /></a></h1>
```

Choisir l'un ou l'autre en commentant/décommentant le code (le code pour l'image est, par défaut, commenté). Les codes de la css seront évidemment à adapter en fonction de la hauteur de votre image (attribut height de #title dans style.css). Concernant la description du blog, si vous affichez une image en titre, il sera nécessaire d'adapter le code css afin de repositionner l'affichage du texte de la description du blog.

Le thème propose plusieurs types de navigation
----------------------------------------------
* un menu reposant sur le plugin menu (css/menus/menuH.css est calibré pour afficher le niveau 2 des sous-catégories sous forme horizontale ; css/menus/menuV.css l'est pour plusieurs niveaux de sous-catégories sous forme verticale), ou sur le plugin SimpleMenu (pas de niveau, css/menus/simplemenu.css)
* un menu fixe situé en bas à droite de la page (inclus dans _footer.html)
* un dock au-dessus du footer listant sous forme d'images les 25 derniers billets (css/dock/dock.css)
* un menu dans le footer ne listant que les catégories de premier niveau et affichant au survol les 80 premiers caractères de leur description

Les slides
-----------
Les slides permettent d'afficher des images originales de 650px de large x 300px de haut.
Ils sont positionnés sous la barre de menu, la liste des billets sera en-dessous et la sidebar à leur droite.
Il n'est pas possible d'afficher simultanément les deux slides dans le blog.

Trois choix possibles :
* Pas de slide
* Slide sans vignette (le titre ainsi que le texte (limité au 300 premiers caractères du billet) s'affiche en bas de l'image sur un fond translucide avec un effet de slide).
* Slide avec vignettes (le titre ainsi que le texte (limité au 100 premiers caractères du billet) s'affiche en bas de l'image sur un fond translucide. La tabulation est sous forme de vignettes).

Par défaut, le slide s'appuie sur les 4 derniers billets sélectionnés. Vous pouvez cependant l'utiliser pour afficher les billets d'une catégorie ou d'un tag.
Pour une catégorie précise on mettra à la place de ```<tpl:Entries selected="1" lastn="4" ignore_pagination="1" no_context="1">``` ceci ```<tpl:Entries category="Url-de-votre-categorie" lastn="4" ignore_pagination="1" no_context="1">```.
Et pour un mot-clé précis, cela ```<tpl:Entries tag="Nom du tag" lastn="4" ignore_pagination="1" no_context="1">```.

Rappel des codes des slides :

slide1.html :

```html
<div id="slider" class="slide1">
  <ul id="sliderContent">
  <tpl:Entries selected="1" lastn="4" ignore_pagination="1" no_context="1">
    <li class="sliderImage">
    {{tpl:EntryImages size="o" html_tag="div" link="entry" length="1"}}
    <span class="bottom"><strong><a href="{{tpl:EntryURL}}">{{tpl:EntryTitle encode_html="1"}}</a></strong>
    <br />
    <!-- # Entry with an excerpt -->
    <tpl:EntryIf extended="1">
    {{tpl:EntryExcerpt encode_html="1" remove_html="1" cut_string="300"}}...
    </tpl:EntryIf>

    <!-- # Entry without excerpt -->
    <tpl:EntryIf extended="0">
    {{tpl:EntryContent encode_html="1" remove_html="1" cut_string="300"}}...
    </tpl:EntryIf>
    </span>
    </li>
  </tpl:Entries>
  <div class="clear sliderImage"></div>
</ul>

<script type="text/javascript">
    $(document).ready(function() {
        $('#slider').s3Slider({
            timeOut: 5000
        });
    });
</script>
</div>
```

slide2.html :

```html
<div class="slide2">
<div class="slide clearfix">
<div class="slide-content">
  <div class="slide-content-post">
  <tpl:Entries selected="1" lastn="4" ignore_pagination="1" no_context="1">
    <div class="post">
      <div class="image">
        {{tpl:EntryImages size="o" html_tag="div" link="entry" length="1"}}
      </div>

      <div class="texte">
        <h2 class="post-title"><a href="{{tpl:EntryURL}}">{{tpl:EntryTitle encode_html="1"}}</a></h2>

        <!-- # Entry with an excerpt -->
        <tpl:EntryIf extended="1">
        <div class="post-content">{{tpl:EntryExcerpt encode_html="1" remove_html="1" cut_string="100"}}...</div>
        </tpl:EntryIf>

        <!-- # Entry without excerpt -->
        <tpl:EntryIf extended="0">
          <div class="post-content">{{tpl:EntryContent encode_html="1" remove_html="1" cut_string="100"}}...</div>
        </tpl:EntryIf>
      </div>

    </div>
  </tpl:Entries>
</div> <!-- End #slide-content-post -->

<div class="slidetabs">
<tpl:Entries selected="1" lastn="4" ignore_pagination="1" no_context="1">
  <div class="vignette">
    {{tpl:EntryImages size="m" html_tag="div" link="entry" length="1"}}
  </div>
</tpl:Entries>
</div>

<script type="text/javascript">
$(".slide2 .slide-content .slidetabs").slidetabs(".slide2 .slide-content .post", {

effect: 'fade',
fadeOutSpeed: "normal",
rotate: true

}).slideshow({ autoplay: true, interval: 4000 });

</script>
</div>
</div>
</div><!-- End #slide-content -->
```

Deux types d'affichage des listes de billets
---------------------------------------------
* affichage single : les billets sont par couple de deux,

```html
<div id="p{{tpl:EntryID}}" class="post single {{tpl:EntryIfOdd}} {{tpl:EntryIfFirst}}" lang="{{tpl:EntryLang}}" role="article">
  <div class="post-meta clearfix">
    <h2 class="post-title left"><a href="{{tpl:EntryURL}}" title="{{tpl:EntryTitle encode_html="1"}}">{{tpl:EntryTitle encode_html="1"}}</a></h2>
  </div><!-- End post-meta -->

  <div class="post-box">
    <div class="post-content">
      <div class="post-intro-moy">

      <!-- # --BEHAVIOR-- publicEntryBeforeContent -->
      {{tpl:SysBehavior behavior="publicEntryBeforeContent"}}

      <div class="entryimages">
      {{tpl:EntryImages size="o" html_tag="div" link="none" legend="none" length="1"}}
      </div>

      <!-- # Entry with an excerpt -->
      <tpl:EntryIf extended="1">
      <p class="moy">{{tpl:EntryExcerpt encode_html="1" remove_html="1" cut_string="300"}} [...]</p>
      </tpl:EntryIf>

      <!-- # Entry without excerpt -->
      <tpl:EntryIf extended="0">
      <p class="moy">{{tpl:EntryContent encode_html="1" remove_html="1" cut_string="300"}} [...]</p>
      </tpl:EntryIf>

      <!-- # --BEHAVIOR-- publicEntryAfterContent -->
      {{tpl:SysBehavior behavior="publicEntryAfterContent"}}

    </div><!-- End post-intro -->
  </div><!-- End post-content -->

  <div class="post-footer clearfix">
    <div class="continue-reading">
      <a href="{{tpl:EntryURL}}" title="{{tpl:lang Continue reading}} {{tpl:EntryTitle encode_html="1"}}">{{tpl:lang Continue reading}}</a>
    </div>
  </div><!-- End post-footer -->
</div><!-- End post-box -->
</div><!-- End post -->

<tpl:EntryIf odd="0"> <div style="clear:left;"> </div> </tpl:EntryIf>
```

* affichage small : sous forme de liste très simplifiée (par exemple dans le fichier archive_month.html).

```html
<tpl:Entries>
  <div class="post">
    <div id="p{{tpl:EntryID}}" class="post-meta clearfix">
      <h2 class="post-title-small left" lang="{{tpl:EntryLang}}"><a
      href="{{tpl:EntryURL}}" title="{{tpl:lang Read}} {{tpl:EntryTitle encode_html="1"}}">{{tpl:EntryTitle encode_html="1"}}</a></h2>
      <p class="post-info right">
        <span class="infopost"><span>{{tpl:lang By}} {{tpl:EntryAuthorLink}}</span>
        {{tpl:EntryDate}}</span>
      </p>
    </div><!-- End post-meta -->
  </div><!-- End post -->
</tpl:Entries>
```

Widget
------
Le contour de chaque widget peut être restreint à son titre, utiliser la class "noborder" prévue à cet effet.

Footer
------
Sont intégrés divers tpl dans le footer (_footer.html) :

* {{tpl:BlogEditor}} : champ "Nom de l'auteur du blog", vous pouvez l'englober par du html (exemple : ```<a href="http://votreurl.ext">nom de votre blog</a>```) à faire depuis l'administration du blog (paramètres du blog) ,
* {{tpl:BlogCopyrightNotice}} : champ "Note de copyright", idem,
* ```{{tpl:lang Designed by}} <a href="https://github.com/brol/breathe">Pierre Van Glabeke</a>``` : mes références d'auteur du thème, merci de le laisser.
* Une zone "A propos" est réservée dans le footer en regard du menu des catégories. Vous êtes libres du titre et du contenu de cette page du moment que son url est ```/pages/A-propos```.

Astuces
-------
* Réduire la navigation dans le contexte billet seul à une catégorie particulière

Installer le plugin [myPostCategoryIf] (http://plugins.dotaddict.org/dc2/details/myPostCategoryIf).

Dans post.html, remplacer le code ```<div id="navlinks">...</div>``` par celui-ci en le renseignant avec le nom urlisé de votre catégorie :

```html
  <tpl:MyPostCategoryIf url="Votre-categorie">
    <div id="navlinks">
      <p><tpl:EntryPrevious restrict_to_category="1"><a href="{{tpl:EntryURL}}"
      title="{{tpl:EntryTitle encode_html="1"}}" class="prev">&#171; {{tpl:EntryTitle encode_html="1"
      cut_string="70"}}</a></tpl:EntryPrevious>
      <tpl:EntryNext restrict_to_category="1"> <span>-</span> <a href="{{tpl:EntryURL}}"
      title="{{tpl:EntryTitle encode_html="1"}}" class="next">{{tpl:EntryTitle encode_html="1"
      cut_string="70"}} &#187;</a></tpl:EntryNext>
      </p>
    </div>
  </tpl:MyPostCategoryIf>
```

* Afficher les sous-catégories sous forme de tableau d'images

Renommer le fichier category.html en category-normal.html et renommer le fichier category-tab.html en category.html.

Décommenter dans style.css la partie "sous-cat en tableau" (lignes 933 - 962).

Dans la description des sous-catégories, il vous faudra alors coder ainsi :

```html
<p><img title="Title de l'image" alt="Texte alternatif de l'image" src="/chemin/vers/image.ext" /></p>
<p class="catdesctxt">Description de la sous-catégorie</p>
```

changelog
v0.6 - 27/12/2022 - Pierre Van Glabeke
* dc2.24 requis

v0.5 - 26/11/2019 - Pierre Van Glabeke
* html5
* héritage de thème, mustek
* dc2.15 requis

0.4 - 12-03-2016 - Pierre Van Glabeke
* ajout jquery 1.11.3
* modif localisation
* modif css pour menu 1.9

0.3 - 19-03-2015 - Pierre Van Glabeke
* correction bug config

0.2 - 18-03-2015 - Pierre Van Glabeke
* html5
* revue complète du code par Nikrou, merci à lui
* modif readme

0.1 - 07-06-2010 - Pierre Van Glabeke
* création du thème

_public.php
<?php
/* BEGIN LICENSE BLOCK
This file is part of breathe, a theme for Dotclear.

Copyright (c) 2013 - 2016
Pierre Van Glabeke https://github.com/brol/breathe

Licensed under the Creative Commons version 4.0 license.
A copy of this license is available in LICENSE file or at
http://creativecommons.org/licenses/by-nc-sa/4.0/deed.fr
END LICENSE BLOCK */

if (!defined('DC_RC_PATH')) { return; }

l10n::set(dirname(__FILE__) . '/locales/' . dcCore::app()->lang. '/public');

dcCore::app()->addBehavior('publicHeadContent','breathePublicHeadContent');

function breathePublicHeadContent()
{
    # appel css menu
	$style = dcCore::app()->blog->settings->themes->breathe_menu;
	if (!preg_match('/^menuH|menuV|simplemenu$/', (string) $style)) {
		$style = 'menuV';
	}

	$theme_url = dcCore::app()->blog->settings->system->themes_url.'/'.dcCore::app()->blog->settings->system->theme;
	echo '<link rel="stylesheet" type="text/css" media="projection, screen" href="'.$theme_url."/css/menus/".$style.".css\" />\n";

    # appel css couleur
    $style = dcCore::app()->blog->settings->themes->breathe_color;
	if (!preg_match('/^gray|spring|summer|autumn|winter$/', (string) $style)) {
		$style = 'gray';
	}

	echo '<link rel="stylesheet" type="text/css" media="screen" href="'.$theme_url."/css/colors/".$style.".css\" />\n";

    # appel css dock
    if (dcCore::app()->blog->settings->themes->breathe_dock=='yesdock')
    {
        echo '<link rel="stylesheet" type="text/css" media="screen" href="'.$theme_url."/css/dock/dock.css\" />\n";
	    echo '<script type="text/javascript" src="'.$theme_url."/js/dock.js\"></script>\n";
	}

    # appel css slide1/slide2 ou aucun
    # appel css slide on the following pages
    if (dcCore::app()->blog->settings->themes->breathe_slide!=0)
    {
        if (dcCore::app()->blog->settings->themes->breathe_slide==1)
        {
            echo '<link rel="stylesheet" type="text/css" media="screen" href="'.$theme_url."/css/slides/slide1.css\" />\n";
           // echo '<script type="text/javascript" src="'.$theme_url."/js/s3Slider.js\"></script>\n";
        }
        else
        {
            echo '<link rel="stylesheet" type="text/css" media="screen" href="'.$theme_url."/css/slides/slide2.css\" />\n";
          //  echo '<script type="text/javascript" src="'.$theme_url."/js/1.11.3/jquery.js\"></script>\n";
            echo '<script type="text/javascript" src="'.$theme_url."/js/jquery.tools.js\"></script>\n";
        }
	}
}

dcCore::app()->tpl->addBlock('BreatheIf', array('tplBreathe', 'BreatheIf'));
dcCore::app()->tpl->addBlock('BreatheIfOnFollowingPages', array('tplBreathe', 'BreatheIfOnFollowingPages'));

class tplBreathe
{
    public static function BreatheIf($attr,$content)
    {
        if (!empty($attr['slide']) && ($attr['slide']==dcCore::app()->blog->settings->themes->breathe_slide))
        {
            return $content;
		}
        if (!empty($attr['dock'])
            && ((strtolower($attr['dock'])=='yes') || ($attr['dock']==1))
            && (dcCore::app()->blog->settings->themes->breathe_dock=='yesdock'))
        {
            return $content;
		}
    }

    public static function BreatheIfOnFollowingPages($attr,$content)
    {
        if (dcCore::app()->url->type=='default-page' && dcCore::app()->blog->settings->themes->breathe_slidenav=='yesslidenav')
        {
            return $content;
        }

    }
}

class behaviorsExcludeCurrentPost
{
	public static function templateBeforeBlock($b,$attr)
	{
	  if ($b == 'Entries' && isset($attr['exclude_current']) && $attr['exclude_current'] == 1)
	  {
		return
		"<?php\n".
		'$params["sql"] = "AND P.post_url != \'".dcCore::app()->ctx->posts->post_url."\' ";'."\n".
		"?>\n";
	  }
	}
}


js/s3Slider.js
/* ------------------------------------------------------------------------
	s3Slider
	
	Developped By: Boban Karišik -> http://www.serie3.info/
        CSS Help: Mészáros Róbert -> http://www.perspectived.com/
	Version: 1.0
	
	Copyright: Feel free to redistribute the script/modify it, as
			   long as you leave my infos at the top.
------------------------------------------------------------------------- */
(function($){  

    $.fn.s3Slider = function(vars) {       
        
        var element     = this;
        var timeOut     = (vars.timeOut != undefined) ? vars.timeOut : 3000;
        var current     = null;
        var timeOutFn   = null;
        var faderStat   = true;
        var mOver       = false;
        var items       = $("#" + element[0].id + "Content ." + element[0].id + "Image");
        var itemsSpan   = $("#" + element[0].id + "Content ." + element[0].id + "Image span");
            
        items.each(function(i) {
    
            $(items[i]).mouseover(function() {
               mOver = true;
            });
            
            $(items[i]).mouseout(function() {
                mOver   = false;
                fadeElement(true);
            });
            
        });
        
        var fadeElement = function(isMouseOut) {
            var thisTimeOut = (isMouseOut) ? (timeOut/2) : timeOut;
            thisTimeOut = (faderStat) ? 10 : thisTimeOut;
            if(items.length > 0) {
                timeOutFn = setTimeout(makeSlider, thisTimeOut);
            } else {
                console.log("Poof..");
            }
        }
        
        var makeSlider = function() {
            current = (current != null) ? current : items[(items.length-1)];
            var currNo      = jQuery.inArray(current, items) + 1
            currNo = (currNo == items.length) ? 0 : (currNo - 1);
            var newMargin   = $(element).width() * currNo;
            if(faderStat == true) {
                if(!mOver) {
                    $(items[currNo]).fadeIn((timeOut/6), function() {
                        if($(itemsSpan[currNo]).css('bottom') == 0) {
                            $(itemsSpan[currNo]).slideUp((timeOut/6), function() {
                                faderStat = false;
                                current = items[currNo];
                                if(!mOver) {
                                    fadeElement(false);
                                }
                            });
                        } else {
                            $(itemsSpan[currNo]).slideDown((timeOut/6), function() {
                                faderStat = false;
                                current = items[currNo];
                                if(!mOver) {
                                    fadeElement(false);
                                }
                            });
                        }
                    });
                }
            } else {
                if(!mOver) {
                    if($(itemsSpan[currNo]).css('bottom') == 0) {
                        $(itemsSpan[currNo]).slideDown((timeOut/6), function() {
                            $(items[currNo]).fadeOut((timeOut/6), function() {
                                faderStat = true;
                                current = items[(currNo+1)];
                                if(!mOver) {
                                    fadeElement(false);
                                }
                            });
                        });
                    } else {
                        $(itemsSpan[currNo]).slideUp((timeOut/6), function() {
                        $(items[currNo]).fadeOut((timeOut/6), function() {
                                faderStat = true;
                                current = items[(currNo+1)];
                                if(!mOver) {
                                    fadeElement(false);
                                }
                            });
                        });
                    }
                }
            }
        }
        
        makeSlider();

    };  

})(jQuery);  


menu.breathe.php
<?php
/* BEGIN LICENSE BLOCK
This file is part of breathe, a theme for Dotclear.

Copyright (c) 2013 - 2016
Pierre Van Glabeke https://github.com/brol/breathe

Licensed under the GPL version 2.0 license.
A copy of this license is available in LICENSE file or at
http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
END LICENSE BLOCK */

# Template style menu
$template_theme_style = array(
	'LI_ACTIVE'			=> 'class="current_page_item"',
	'LI_LAST'			=> 'class="last_menu"',
	'LI_ACTIVE_LAST'	=> 'class="current_page_item last_menu"',
	'LI_PARENT'			=> 'class="top_parent"',
	'LI_FIRST_PARENT'	=> 'class="first_menu top_parent"',
	'LI_LAST_PARENT'	=> 'class="last_menu top_parent"',
	'LI_ACTIVE_FIRST_PARENT' 	=> 'class="first_menu top_parent"',
	'LI_ACTIVE_LAST_PARENT' 	=> 'class="last_menu top_parent"',
	'LI_SMPARENT'		=> 'class="current_page_parent"',
	'LI_ACTIVE_SM'		=> 'class="current_page_item"'
);

# Widget style menu
$widget_theme_style = array(
	'W_LI_ACTIVE'		=> 'class="current_page_item"',
	'W_LI_ACTIVE_SM'	=> 'class="current_page_item"'
);


_config.php
<?php
/* BEGIN LICENSE BLOCK
This file is part of breathe, a theme for Dotclear.

Copyright (c) 2013 - 2015
Pierre Van Glabeke https://github.com/brol/breathe

Licensed under the Creative Commons version 4.0 license.
A copy of this license is available in LICENSE file or at
http://creativecommons.org/licenses/by-nc-sa/4.0/deed.fr
END LICENSE BLOCK */

if (!defined('DC_CONTEXT_ADMIN')) { return; }

//PARAMS

# Translations
l10n::set(__DIR__ . '/locales/' . dcCore::app()->lang . '/main');

# Default values
$default_menu = 'menuV';
$default_color = 'gray';
$default_dock = 'yesdock';
$default_slidenav = 'yesslidenav';
$default_slide = 0;

# Settings
$my_menu = dcCore::app()->blog->settings->themes->breathe_menu;
$my_color = dcCore::app()->blog->settings->themes->breathe_color;
$my_dock = dcCore::app()->blog->settings->themes->breathe_dock;
$my_slidenav = dcCore::app()->blog->settings->themes->breathe_slidenav;
$my_slide = dcCore::app()->blog->settings->themes->breathe_slide;

# Menu type
$breathe_menu_combo = array(
	__('Menu (horizontal)') => 'menuH',
	__('Menu (vertical)') => 'menuV',
	__('SimpleMenu') => 'simplemenu'
);

# Color scheme
$breathe_color_combo = array(
	__('Default') => 'gray'
  //,
	//__('Spring') => 'spring',
//	__('Summer') => 'summer',
//	__('Autumn') => 'autumn',
//	__('Winter') => 'winter'
);

# dock
$breathe_dock_combo = array(
	__('Yes') => 'yesdock',
	__('No') => 'nodock'
);

$html_fileslide = array(); $html_contentslide = array();

# slide1
$html_fileslide[1] = path::real(dcCore::app()->blog->themes_path).'/'.dcCore::app()->blog->settings->system->theme.'/tpl/_slide1.html';
if (!is_writable(dirname($html_fileslide[1]))) {
    throw new Exception(
        sprintf(__('File %s does not exist and directory %s is not writable.'),
                $html_fileslide[1],dirname($html_fileslide[1]))
    );
}
$html_contentslide[1] = file_get_contents($html_fileslide[1]);

# slide2
$html_fileslide[2] = path::real(dcCore::app()->blog->themes_path).'/'.dcCore::app()->blog->settings->system->theme.'/tpl/_slide2.html';
if (!is_writable(dirname($html_fileslide[2]))) {
	throw new Exception(
		sprintf(__('File %s does not exist and directory %s is not writable.'),
                    $css_fileslide[2],dirname($html_fileslide[2]))
            );
}
$html_contentslide[2] = file_get_contents($html_fileslide[2]);

# slide on the following pages
$breathe_slidenav_combo = array(
	__('Yes') => 'yesslidenav',
	__('No') => 'noslidenav'
);

// POST ACTIONS

if (!empty($_POST))
{
	try
	{
		dcCore::app()->blog->settings->addNamespace('themes');

		# Menu type
		if (!empty($_POST['breathe_menu']) && in_array($_POST['breathe_menu'],$breathe_menu_combo))
		{
			$my_menu = $_POST['breathe_menu'];

		} elseif (empty($_POST['breathe_menu']))
		{
			$my_menu = $default_menu;

		}
		dcCore::app()->blog->settings->themes->put('breathe_menu',$my_menu,'string','Menu to display',true);

		# Color scheme
		if (!empty($_POST['breathe_color']) && in_array($_POST['breathe_color'],$breathe_color_combo))
		{
			$my_color = $_POST['breathe_color'];


		} elseif (empty($_POST['breathe_color']))
		{
			$my_color = $default_color;

		}
		dcCore::app()->blog->settings->themes->put('breathe_color',$my_color,'string','Color display',true);

		# Dock scheme
		if (!empty($_POST['breathe_dock']) && in_array($_POST['breathe_dock'],$breathe_dock_combo))
		{
			$my_dock = $_POST['breathe_dock'];


		} elseif (empty($_POST['breathe_dock']))
		{
			$my_dock = $default_dock;

		}
		dcCore::app()->blog->settings->themes->put('breathe_dock',$my_dock,'string','Dock display',true);

		# Slide scheme
		if (!empty($_POST['breathe_slide']) && ($_POST['breathe_slide']==1 || $_POST['breathe_slide']==2))
		{
			$my_slide = $_POST['breathe_slide'];

            if (isset($_POST['slide'.$_POST['breathe_slide']]))
            {
                $html_contentslide[$_POST['breathe_slide']] = $_POST['slide'.$_POST['breathe_slide']];
                @$fp = fopen($html_fileslide[$_POST['breathe_slide']],'wb');
                fwrite($fp,$html_contentslide[$_POST['breathe_slide']]);
                fclose($fp);
            }
		} else
		{
			$my_slide = $default_slide;
		}
		dcCore::app()->blog->settings->themes->put('breathe_slide',$my_slide,'integer', 'Display slide',true);

		# Slide on the following pages scheme
		if (!empty($_POST['breathe_slidenav']) && in_array($_POST['breathe_slidenav'],$breathe_slidenav_combo))
		{
			$my_slidenav = $_POST['breathe_slidenav'];
		} elseif (empty($_POST['breathe_slidenav']))
		{
			$my_slidenav = $default_slidenav;
		}
		dcCore::app()->blog->settings->themes->put('breathe_slidenav',$my_slidenav,'string','Slidenav display',true);

		// Blog refresh
		dcCore::app()->blog->triggerBlog();

		// Template cache reset
		dcCore::app()->emptyTemplatesCache();

		dcPage::success(__('Theme configuration has been successfully updated.'),true,true);
	}
	catch (Exception $e)
	{
		dcCore::app()->error->add($e->getMessage());
	}
}

// DISPLAY

# Menu type
echo
'<div class="fieldset"><h4>'.__('Menu').'</h4>'.
'<p class="field"><label>'.__('Menu to display:').'</label>'.
form::combo('breathe_menu',$breathe_menu_combo,$my_menu).
'</p>'.
'<p class="info">'.__('Plugins menu allowed: <a href="http://plugins.dotaddict.org/dc2/details/menu">Menu</a> plugin (horizontal and vertical), or SimpleMenu.').'</p>'.
'</div>';

# Color scheme
echo
'<div class="fieldset"><h4>'.__('Colors').'</h4>'.
'<p class="field"><label>'.__('Color display:').'</label>'.
form::combo('breathe_color',$breathe_color_combo,$my_color).
'</p>'.
'</div>';

# Dock scheme
echo
'<div class="fieldset"><h4>'.__('Dock').'</h4>'.
'<p class="field"><label>'.__('Dock display:').'</label>'.
form::combo('breathe_dock',$breathe_dock_combo,$my_dock).
'</p>'.
'<p class="info">'.__('To display pictures in the dock, you must have installed the plugin <a href="http://plugins.dotaddict.org/dc2/details/listImages">ListImages</a>.').'</p>'.
'</div>';

# slides
echo
'<div class="fieldset"><h4>'.__('Slides').'</h4>'.
'<p class="info">'.__('To display pictures in a slide, you must have installed the plugin <a href="http://plugins.dotaddict.org/dc2/details/listImages">ListImages</a>.<br />The slides can display originals images 650px wide x 300px high.<br />They are positioned under the menu bar, the range of tickets will be below and sidebar to the right.<br />It is not possible to simultaneously view two slides in the blog.').'</p>'.
'<p>'.
	form::radio(array('breathe_slide','breathe_slide0'),0,($my_slide==0)).
	'<label class="classic" for="breathe_slide0">'.
		__('No slide').
	'</label>'.
'</p>'.
'<p>'.
	form::radio(array('breathe_slide','breathe_slide1'),1,($my_slide==1)).
	'<label class="classic" for="breathe_slide1">'.
		__('Slide without Thumbnails').
	'</label>'.
'</p>'.
'<p class="info">'.__('The title and the text (limited to the first 300 characters of the ticket) appears at the bottom of the image on a translucent background with a slide effect.').'</p>';

echo
'<p class="area"><label for="slide1">'.__('Code:').' '.
form::textarea('slide1',60,20,html::escapeHTML($html_contentslide[1])).'</label></p>'.
'<p>'.
	form::radio(array('breathe_slide','breathe_slide2'),2,($my_slide==2)).
	'<label class="classic" for="breathe_slide2">'.
		__('Slide with Thumbnails').
	'</label>'.
'</p>'.
'<p class="info">'.__('The title and the text (limited to the first 100 characters of the ticket) appears at the bottom of the image on a translucent background. The Tab is as thumbnails.').'</p>';

echo
'<p class="area"><label for="slide2">'.__('Code:').' '.
form::textarea('slide2',60,20,html::escapeHTML($html_contentslide[2])).'</label></p>'.
'<p class="info">'.__('By default, the slide is based on the last 4 selected tickets. However, you can use it to display the Notes of a category or tag.<br />For a specific class will be put in place of <code>&lt;tpl:Entries selected="1" lastn="4" ignore_pagination="1" no_context="1"&gt;</code> this <code>&lt;tpl:Entries category="Url-of-the-category" lastn="4" ignore_pagination="1" no_context="1"&gt;</code>.<br />And for a specific tag, this <code>&lt;tpl:Entries tag="Name of the tag" lastn="4" ignore_pagination="1" no_context="1"&gt;</code>.').'</p>';

# slide on the following pages
echo
'<p class="field"><label>'.__('Display on the following pages:').'</label>'.
form::combo('breathe_slidenav',$breathe_slidenav_combo,$my_slidenav).
'</p>'.
'</div>';